metapool/prep.py

```diff
@@ -0,0 +1,424 @@
+    ----------
+    sheet: sample_sheet.SampleSheet
+        Object from where to extract the data.
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame object with the sample information.
+    """
+
+    # used to get the right order every time
+    columns = list(sheet.samples[0].keys())
+
+    data = []
+    for sample in sheet.samples:
+        data.append([sample[column] for column in columns])
+
+    out = pd.DataFrame(data=data, columns=[c.lower() for c in columns])
+
+    return out.set_index('sample_id')
+
+
+def is_nonempty_gz_file(name):
+    """Taken from https://stackoverflow.com/a/37878550/379593"""
+    with gzip.open(name, 'rb') as f:
+        try:
+            file_content = f.read(1)
+            return len(file_content) > 0
+        except Exception:
+            return False
+
+
+def remove_qiita_id(project_name):
+    # project identifiers are digit groups at the end of the project name
+    # preceded by an underscore CaporasoIllumina_550
+    qiita_id_re = re.compile(r'(.+)_(\d+)$')
+
+    # no matches
+    matches = re.search(qiita_id_re, project_name)
+    if matches is None:
+        return project_name
+    else:
+        # group 1 is the project name
+        return matches[1]
+
+
+def get_run_prefix(run_path, project, sample, lane, pipeline):
+    """For a sample find the run prefix
+
+    Parameters
+    ----------
+    run_path: str
+        Base path for the run
+    project: str
+        Name of the project
+    sample: str
+        Sample name
+    lane: str
+        Lane number
+    pipeline: str
+        The pipeline used to generate the data. Should be one of
+        `atropos-and-bowtie2` or `fastp-and-minimap2`.
+
+    Returns
+    -------
+    str
+        The run prefix of the sequence file in the lane, only if the sequence
+        file is not empty.
+    """
+    base = os.path.join(run_path, project)
+    path = base
+
+    # each pipeline sets up a slightly different directory structure,
+    # importantly fastp-and-minimap2 won't save intermediate files
+    if pipeline == 'atropos-and-bowtie2':
+        qc = os.path.join(base, 'atropos_qc')
+        hf = os.path.join(base, 'filtered_sequences')
+
+        # If both folders exist and have sequence files always prefer the
+        # human-filtered sequences
+        if _exists_and_has_files(qc):
+            path = qc
+            if _exists_and_has_files(hf):
+                path = hf
+    elif pipeline == 'fastp-and-minimap2':
+        qc = os.path.join(base, 'trimmed_sequences')
+        hf = os.path.join(base, 'filtered_sequences')
+
+        if _exists_and_has_files(qc) and _exists_and_has_files(hf):
+            path = hf
+        elif _exists_and_has_files(qc):
+            path = qc
+        elif _exists_and_has_files(hf):
+            path = hf
+        else:
+            path = base
+    else:
+        raise ValueError('Invalid pipeline "%s"' % pipeline)
+
+    results = glob(os.path.join(path, '%s_S*_L*%s_R*.fastq.gz' %
```

Comment:
Not important but perhaps nice for future iterations:
`'%s_S*_L*%s_R*.fastq.gz' % (sample, lane)`
can be
`f'{sample}_S*_L*{lane}_R*.fastq.gz'`

Reply:
👍  I need to get on with the times and jump on the f-string bandwagon.